src/dealWizard/components/proposalLeadSection/proposalLeadSection.ts

```diff
@@ -1,9 +1,38 @@
 import { bindable } from "aurelia-typed-observable-plugin";
 import { IProposalLead } from "entities/DealRegistrationTokenSwap";
 import { WizardErrors } from "services/WizardService";
+import "./proposalLeadSection.scss";
+import { autoinject, bindingMode } from "aurelia-framework";
+import { EthereumService } from "../../../services/EthereumService";
+import { EventAggregator, Subscription } from "aurelia-event-aggregator";
 
+@autoinject
 export class ProposalLeadSection {
   @bindable errors: WizardErrors<IProposalLead>;
-  @bindable data: Record<string, string>;
+  @bindable({defaultBindingMode: bindingMode.twoWay}) address: string;
+  @bindable({defaultBindingMode: bindingMode.twoWay}) email?: string;
   @bindable disabled = false;
+
+  ethAddress?: string;
+
+  private accountSubscription: Subscription;
+
+  constructor(
+    private eventAggregator: EventAggregator,
+    private ethereumService: EthereumService,
+  ) {
+    this.ethAddress = this.ethereumService.defaultAccountAddress;
+    this.accountSubscription = this.eventAggregator.subscribe("Network.Changed.Account", address => {
+      this.ethAddress = address;
+    });
+
+  }
+
+  async connectToWallet() {
+    await this.ethereumService.connect();
```

Comment:
no need for await here.  Doesn't behave as you might expect, anyways

Reply:
It's good practice to await promises usually. (and my IDE complains about it. I just wanted to get rid of the warning actually 🤣 )

What do you mean about "Doesn't behave as you might expect" ?